### DISCUSSION TITLE
Using custom YAML metadata in partials

### DISCUSSION THREAD

**Discussion by pjotrhey** [quarto-dev/quarto-cli#11652] [Q&A] [answered]
### Description

Hi all, 

I cannot find how I can use custom YAML metadata with partials. Below is a very minimal example. It does output the aﬀiliation name (with is a standard Quarto YAML field) but not the fullname (which is not a standard name).

This is my index.qmd
````qmd
---
title: "Min Example"
affiliations:
  - id: hms
    name: YAML Name
    fullname: YAML Fullname
format: 
  pdf: 
    template-partials:
      - "before-body.tex"
editor: source
---
````

And then this my before-body.tex:
````
$for(by-affiliation)$%
The affiliation name is: $it.name$
The fullname is: $it.fullname$
$endfor$
````

Thanks in advance!

**Comment by mcanouil** [comment]
Authors and affiliations follow a specific schema, you cannot add whatever you want in it.

https://quarto.org/docs/journals/authors.html

**Reply by pjotrhey** [reply]
Thanks for your answer, but is there no workaround here? From that page it is not clear for me whether there may be a workaround (or not).

**Reply by mcanouil** [reply]
What do you mean by workaround?
You want to work around the standardised schema? (If so, then define your own `my-affiliations` instead of `affiliations` and write all logics in your template/partials, but I would not recommend this if you intend to share with other Quarto users because you'll likely confuse them with your schema)

`name` is supposed to be the display name usually the full name.

**Reply by pjotrhey** [reply]
A workaround to make the proposed YAML and intended use to work. This to make it work for a preprint for a scientific article where the reference name is: Department, Faculty, University, City, Country, Research Institute. As Faculty and Research Institute are not part of the standardised affiliation schema I cannot make this work. So I was thinking about adding a custom key to aﬀiliations which I then can include in my partials. 
Defining my own key (like my-aﬀiliations) will not make it work anymore with attaching the aﬀiliation(s) to the authors: 
````qmd
author: 
    aﬀiliations: 
      - ref: hms
````

**Reply by mcanouil** [reply, marked as answer]
You cannot modify the current schema.
Your only way (as far as I know) is to use another YAML key and as I said implement all the logics in your partials (possibly with the help of a LUA filter).

**Reply by pjotrhey** [reply]
Thanks, thats what I was afraid of. Atm, I don't understand LUA filters (yet). ;)

**Reply by mcanouil** [reply]
Oh, an ugly workaround could be to use another key from the affiliation schema if you don't use them all.
Since you are using one format and partials/templates, you should not face side effects.
For example, you could use `url` as your `fullname` key.

https://quarto.org/docs/journals/authors.html#affiliations-schema
